### ISSUE TITLE
[CRITICAL] Error while checking job templates HINT: check the CUSTOM_DIRECTIVE syntax

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#999]
In GitLab by @orioltinto on Mar 27, 2023, 12:07

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

4.0.76

### Summary

What is the proper syntax for the CUSTOM_DIRECTIVES of a platform?

I used the syntax from [the documentation](https://autosubmit.readthedocs.io/en/master/userguide/configure/develop_a_project.html#detailed-platform-configuration) (i.e. `CUSTOM_DIRECTIVES: ["#SBATCH -p small", "#SBATCH --no-requeue", "#SBATCH --usage"]`) and it raises this error:

```
 [CRITICAL] Error while checking job templates HINT: check the CUSTOM_DIRECTIVE syntax in your jobs configuration files. [eCode=7014]
```

If I use a single element without [] it works, but I need to use several directives.

What is the expected syntax now?

()

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 27, 2023, 12:09

Hello @orioltinto ,

Sorry, It should be `CUSTOM_DIRECTIVES: "['#SBATCH -p small', '#SBATCH --no-requeue', '#SBATCH --usage']"`

I'll update it

**Comment by kinow** [comment]
In GitLab by @orioltinto on Mar 27, 2023, 12:41

Hi @dbeltrankyl, thanks for the answer, it works.


However, I don't know the details of the implementation and the reason of using a single string, but as a user it would be much clearer to use the yaml syntax for lists.

```yaml
CUSTOM_DIRECTIVES: ["#SBATCH -p small", "#SBATCH --no-requeue", "#SBATCH --usage"]

```

or

```yaml
CUSTOM_DIRECTIVES:
   -"#SBATCH -p small"
   -"#SBATCH --no-requeue"
   -"#SBATCH --usage"

```

Is there a strong reason against for not doing that?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 27, 2023, 12:50

No, there is no strong reason. It is a pending task.

These parameters come from Autosubmit 3 ini files. The person who did this development integrated a json_load to load the info stored in that parameter. 

This shouldn't be necessary anymore with YAML files, but we did not solve that yet and we still using the same method that was used in INI files.

In fact I created #1000 to solve similar issues like this one
